utils/githooks/commit-msg.d/10-watermark.py

```diff
@@ -30,6 +30,15 @@ def find_hooks():
     return hooks
 
 
+def emit_watermark(msg):
+    """Print the Required-githooks value"""
+    msg.write("Required-githooks: true\n")
+    skipped = os.environ.get("DAOS_SKIP", None)
+    if skipped:
+        msg.write(f"Skipped-githooks: {skipped}\n")
+    msg.write("\n")
```

Comment:
FWIW, I think we should phase out the Required-githooks watermark. They're really just a convenience, since we have GHA to do the same checks. The only one missing (I think) is the copyright githook doesn't have a GHA.

There are holes in the watermark, like:
1. Users can just manually paste the watermark. As a gatekeeper, I see this A LOT.
2. It doesn't get added in every scenario. E.g. some merges.
3. Even this new "Skipped-githooks" wouldn't be added if you use `--amend`

Thoughts @brianjmurrell?

Reply:
Honestly, this would be fine with me but we should fix the copyright issue and check that in gha

Reply:
Agree. Beyond the scope of this PR anyway, but I guess where I was going with that is that if @brianjmurrell agrees on phasing out the watermark, you wouldn't need to bother touching it

Reply:
> FWIW, I think we should phase out the Required-githooks watermark. They're really just a convenience, since we have GHA to do the same checks. The only one missing (I think) is the copyright githook doesn't have a GHA.

Should we fix that before removing the watermark?  `Jenkinsfile` would need updating if/when removing that.

> There are holes in the watermark, like:
> 
> 1. Users can just manually paste the watermark. As a gatekeeper, I see this A LOT.

I would liken this to faking test results to get your PR landed and treated as such (with whatever kind of consequences that might have).  Not using them does not fail your PR so there is on reason to fake them.

> 2. It doesn't get added in every scenario. E.g. some merges.

That would be a bug that needs fixing, assuming we were keeping them.

> 3. Even this new "Skipped-githooks" wouldn't be added if you use `--amend`

This is a general problem of having a facility to carry pragams forward in commits.  I have some hackery around that (but it's rough and fails all kinds of corner cases) as does @ashleypittman I believe.  I wonder how fail-proof his is.

Reply:
Yes, I do think we need a copyright GHA before removing the watermark.
As for faking Required-githooks, I think think the problem is that the watermark so often doesn't get added, or the githooks themselves fail for various reasons. So I suspect people have just added the watermark to their standard copy-paste commit pragmas so they don't see the stage warning. It's hard to enforce not faking it because (one example):
You could modify a lot of files and use `--amend --no-verify` and the hook would still be there from the first commit. This one is nearly impossible to enforce because if you use `--no-verify` the hooks don't run. So no amount of intelligent hooks can work around this.

I think the core problem is that we're effectively relying on a client-side check to tell the server (GitHub/us) that something is correct. That's why I think removing the watermark entirely and having a corresponding GHA for each hook is the proper solution.


> This is a general problem of having a facility to carry pragams forward in commits. I have some hackery around that (but it's rough and fails all kinds of corner cases)

According to [githooks.com](https://githooks.com/), there are 17 hooks, most of which we don't use. To account for most corner cases, I think we would need to break the githooks into utility files and write duplicate hooks for the various hook points.